assets/express/src/api/user/index.ts

```diff
@@ -0,0 +1,14 @@
+import { ModuleFactory } from '@common';
+import { userServiceFactory } from '@services';
+import { UserController } from './interfaces';
+import { userControllerFactory } from './v1';
+
+export * from './interfaces'
+
+export const userModuleFactory: ModuleFactory<
+  UserController
+> = function ({ userRepoitory }) {
+  const userService = userServiceFactory(userRepoitory);
```

Comment:
When we introduce DI, here instead of passing just the `userRepository` we need to provide all the possibilities. In real life, we would need different services and repositories basically everywhere. Sometimes you need to use an already implemented service method of another service or a repository, so you can implement the business logic.